torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1548,16 +1548,58 @@ def _update_divisible(self):
 
         self.divisible = new_divisible
 
+    @_lru_cache
+    def _is_divisible(self, x, y):
+        return x % y == 0 or x % y in self.divisible
+
+    @_lru_cache
+    def _elim_floor(self, x, y):
+        if type(x) is sympy.floor and type(x.args[0]) is sympy.Mul:
+            m1, m2 = x.args[0].args
+            ry = 1 / y
+            if m2 == ry and self._is_divisible(m1, y):  # inner mul is div
```

Comment:
Yes. But maybe I'm missing something? To optimize `floor(m1 / m2) * y` to `m1`:
1. `m1` and `m2` must be divisible
2. `m2` must be equal to `y`.

But SymPy represents this division as multiplication:
 ```
 [nav] In [7]: sympy.srepr(sympy.floor(x / y) * y)
Out[7]: "Mul(Symbol('y'), floor(Mul(Symbol('x'), Pow(Symbol('y'), Integer(-1)))))"
 ```
 
So we first check (2) but use `1/y` because `m2` is `1/something`.
Then we check (1) but against `y` and not `m2` because `m2` is `1/something`.